ludwig/encoders/text_encoders.py

```diff
@@ -68,17 +87,47 @@ def __init__(
             )
             sys.exit(-1)
 
-        self.transformer = BertModel.from_pretrained(
-            pretrained_model_name_or_path
-        )
+        if use_pretrained:
+            self.transformer = BertModel.from_pretrained(
+                pretrained_model_name_or_path
+            )
+            if trainable:
+                self.transformer.train()
+        else:
+            config = BertConfig(
+                vocab_size=vocab_size,
+                hidden_size=hidden_size,
+                num_hidden_layers=num_hidden_layers,
+                num_attention_heads=num_attention_heads,
+                intermediate_size=intermediate_size,
+                hidden_act=hidden_act,
+                hidden_dropout_prob=hidden_dropout_prob,
+                attention_probs_dropout_prob=attention_probs_dropout_prob,
+                max_position_embeddings=max_position_embeddings,
+                type_vocab_size=type_vocab_size,
+                initializer_range=initializer_range,
+                layer_norm_eps=layer_norm_eps,
+                pad_token_id=pad_token_id,
+                gradient_checkpointing=gradient_checkpointing,
+                position_embedding_type=position_embedding_type,
+                use_cache=use_cache,
+                classifier_dropout=classifier_dropout
+            )
+            self.transformer = BertModel(config)
+
         self.reduce_output = reduce_output
         if not self.reduce_output == 'cls_pooled':
             self.reduce_sequence = SequenceReducer(reduce_mode=reduce_output)
```

Comment:
As reduce_output can be None (on 'none' or 'None'), we want to make sure that the output_shape takes that into consideration, and contains an if like:
```
if self.reduce_output is None:
  return torch.Size([self.max_sequence_length, self.transformer.embedding_size])
```

Reply:
Thanks for the input on slack -- updated with the fix!